src/ContentMessages.js

```diff
@@ -283,7 +318,88 @@ class ContentMessages {
         });
     }
 
-    sendContentToRoom(file, roomId, matrixClient) {
+    getUploadLimit() {
+        if (this._mediaConfig !== null && this._mediaConfig["m.upload.size"] !== undefined) {
+            return this._mediaConfig["m.upload.size"];
+        }
+    }
+
+    async sendContentListToRoom(files, roomId, matrixClient) {
+        if (matrixClient.isGuest()) {
+            dis.dispatch({action: 'require_registration'});
+            return;
+        }
+
+        const isQuoting = Boolean(RoomViewStore.getQuotingEvent());
+        if (isQuoting) {
+            const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+            const shouldUpload = await new Promise((resolve) => {
+                Modal.createTrackedDialog('Upload Reply Warning', '', QuestionDialog, {
+                    title: _t('Replying With Files'),
+                    description: (
+                        <div>{_t(
+                            'At this time it is not possible to reply with a file ' +
+                            'so this will be sent without being a reply.',
```

Comment:
we're asking the user a question (continue/cancel), but this comes as a statement which implies that the buttons mean nothing. Maybe something like `would you like to upload this file without replying?`